app/services/link_creator.rb

```diff
@@ -0,0 +1,22 @@
+class LinkCreator < ApplicationService
+  def initialize(params, user)
+    @params = params
+    @user = user
+  end
+
+  def call
+    link = find_by_hash(@params[:url_hash])
+    link.present? ? link : create
+  end
+
+  private
+
+  def find_by_hash(hash)
+    links = Link.find_by_url_hash(hash)
+    links.present? ? links[0] : nil
```

Comment:
Why is `links[0]` here? `links` shouldn't be a collection. `find_by_` methods return first found record or `nil`.
So you can just rewrite it like this:
```
def find_by_hash(hash)
  Link.find_by_url_hash(hash)
end
```